### PR TITLE
269: added tests for Participant Queryset

### PR DESCRIPTION
**Pull request by asgharsir**

### CODE REVIEW

**Comment by asgharsir** [comment]
PR #274

File: participant/tests/test_querysets.py [hunk 1]
```diff
@@ -45,3 +45,60 @@ def test_when_user_is_not_user_instance(self):
...
+        other_organization = OrganizationFactory(name="other_organization")
+        self.participant4 = ParticipantFactory(organization=other_organization)
+
+        self.user = UserFactory(
+            username="user_with_organization", organization=organization
+        )
+        self.superuser = UserFactory(
+            organization=organization, is_superuser=True, username="test_superuser"
+        )
+        self.user_without_participant = UserFactory(
```
Comment (fericsepi) [inline participant/tests/test_querysets.py]: Please rename to `user_without_organisation`

File: participant/tests/test_querysets.py [hunk 2]
```diff
@@ -45,3 +45,60 @@ def test_when_user_is_not_user_instance(self):
...
+            username="user_without_organization"
+        )
```
Comment (fericsepi) [inline participant/tests/test_querysets.py]: loose the `username`-s please

File: participant/tests/test_querysets.py [hunk 3]
```diff
@@ -45,3 +45,60 @@ def test_when_user_is_not_user_instance(self):
...
+
+        organization_participants = [
+            self.participant1,
+            self.participant2,
+            self.participant3,
+        ]
+
+        for participant in organization_participants:
+            with self.subTest(participant_organization=participant.organization.name):
+                self.assertTrue(participant in participants)
```
Comment (fericsepi) [inline participant/tests/test_querysets.py]: You'd want to check that using `filter_by_user()` actually returns what you'd expect.
⚠️ Please also note the function name change (I feel that would be more descriptive).

```python
def test_participant_filter_by_user_with_organisation(self):
    # Participant 1, 2 and 3 took the same test that belong to the Organisation of self.user
    expected_participants = [self.participant1, self.participant2, self.participant3]
    filtered_participants = Participants.objects.filter_by_user(user=self.user)

    self.assertEqual(expected_participants, list(filtered_participants))
```

File: participant/tests/test_querysets.py [hunk 4]
```diff
@@ -45,3 +45,60 @@ def test_when_user_is_not_user_instance(self):
...
+
+    def test_when_superuser(self):
+        all_participants = Participant.objects.all()
+        participants = Participant.objects.filter_by_user(user=self.superuser)
+
+        self.assertEqual(list(all_participants), list(participants))
+
+    def test_when_user_without_participant(self):
```
Comment (fericsepi) [inline participant/tests/test_querysets.py]: rename test to `test_filter_by_user_without_organisation`

File: participant/tests/test_querysets.py (hunk 3 above)
Comment (asgharsir) [reply participant/tests/test_querysets.py]: I already tried this:
def test_participant_filter_by_user_with_organisation(self):
    participants = Participant.objects.filter_by_user(user=self.user)

        organization_participants = [
            self.participant1,
            self.participant2,
            self.participant3,
        ]
        self.assertEqual(organization_participants, list(participants))

BUT IT SURPRISED ME WITH THE FOLLOWING:
AssertionError: Lists differ: [<Par[18 chars]ant: 6da72b3b-dff5-409e-9a25-5f83583af845 For:[175 chars] 39>] != [<Par[18 chars]ant: 34f66544-173c-48da-a990-a6f7fd87d973 For:[175 chars] 39>]

I actually copied it, to know WHY?

Comment (asgharsir) [reply participant/tests/test_querysets.py]: I know this, but the question is why other code is working without sort.

Comment (asgharsir) [reply participant/tests/test_querysets.py]: Oh no. That one is also not working now. Unlucky me. It worked only once and I put you the question. Random can be that bad I never know.

Comment (asgharsir) [reply participant/tests/test_querysets.py]: sorted instances were not supported, so subtest was the most efficient. Or a count. Others are expensive.

Comment (fericsepi) [reply participant/tests/test_querysets.py]: oh yeah, try this instead:
```python
def test_org_filter_user_with_participant(self):
    # Participant 1, 2 and 3 took the same test that belong to the Organisation of self.user
    expected_ids = [self.participant1.id, self.participant2.id, self.participant3.id]
    filtered_ids = Participant.objects.filter_by_user(user=self.user).values_list('id', flat=True)

    self.assertEqual(set(expected_ids), set(filtered_ids))
```

Comment (asgharsir) [reply participant/tests/test_querysets.py]: good idea. if subtest be avoided

**Pull request merged**